subsys/bluetooth/services/Kconfig.dis

```diff
@@ -0,0 +1,109 @@
+# Kconfig - Bluetooth GATT Device Information service
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig BT_GATT_DIS
+	bool "Enable BLE GATT Device Information service"
+	default n
+
+if (BT_GATT_DIS)
+
+choice
+	prompt "Device Information service logging level"
+	default BT_GATT_DIS_LOG_LEVEL_INF
+
+ config BT_GATT_DIS_LOG_LEVEL_OFF
+	bool "Off"
+config BT_GATT_DIS_LOG_LEVEL_ERR
+	bool "Error"
+config BT_GATT_DIS_LOG_LEVEL_WRN
+	bool "Warning"
+config BT_GATT_DIS_LOG_LEVEL_INF
+	bool "Info"
+config BT_GATT_DIS_LOG_LEVEL_DBG
+	bool "Debug"
+
+ endchoice
+
+ config BT_GATT_DIS_LOG_LEVEL
+	int
+	default 0 if BT_GATT_DIS_LOG_LEVEL_OFF
+	default 1 if BT_GATT_DIS_LOG_LEVEL_ERR
+	default 2 if BT_GATT_DIS_LOG_LEVEL_WRN
+	default 3 if BT_GATT_DIS_LOG_LEVEL_INF
+	default 4 if BT_GATT_DIS_LOG_LEVEL_DBG
+
+
+config BT_GATT_DIS_MODEL
+	string "Model name"
+	default SOC
+	help
+	 The device model inside Device Information Service.
+
+config BT_GATT_DIS_MANUF
+	string "Manufacturer name"
+	default "Manufacturer"
+	help
+	 The device manufacturer inside Device Information Service.
+
+config BT_GATT_DIS_PNP
+	bool "Enable PnP_ID characteristic"
```

Comment:
I would probably default this to y since usually PID and VID is used to probe a driver, at least in Linux it works that way. That said if we have good defaults maybe we should remove this and always have the PNP char declared.

Reply:
PnP is optional characteristic. From the top of my mind it seems that it is mandatory only for HOG. If we think about DIS globally t should not be enabled by default.

Reply:
But the other information DIS carries can easily be mapped to device name, e.g: manufacturer and model, while PNP is only provided by DIS which is probably why HoG mandates it. So if you are going to use DIS Id assume it is for enabling the use of PID/VID, not to access manufacturer name, etc.

Reply:
I guess we can have PNP enabled by default, because it is still optional charactertistic for HRP (especially when it improves interoperability). However, I would still leave it configurable and possibly make all DIS characteristics configurable

Reply:
I think we should make all attributes configurable then, and default to off. Other services can then select the characteristics it mandates.

Reply:
Off by default.